### PR TITLE
Fix slider repeats playing sample twice

### PR DESCRIPTION
**Pull request by Shawdooow**
they are no longer glued to the end circle only + sound fix (their sound was being played twice and was noisey)

### CODE REVIEW

File: osu.Game.Rulesets.Osu/Objects/Slider.cs [hunk 1]
```diff
@@ -155,13 +155,20 @@ private void createRepeatPoints()
             var repeatPointDistance = Math.Min(Distance, length);
             var repeatDuration = length / Velocity;
 
+            bool sliderStart = true;
```
Comment (kmoore43) [inline osu.Game.Rulesets.Osu/Objects/Slider.cs]: Since this is local, I don't see it being set anywhere else. If that's the case is it required?

File: osu.Game.Rulesets.Osu/Objects/Slider.cs [hunk 2]
```diff
@@ -155,13 +155,20 @@ private void createRepeatPoints()
...
             {
+                sliderStart = !sliderStart;
+
                 for (var d = repeatPointDistance; d <= length; d += repeatPointDistance)
                 {
                     var repeatStartTime = StartTime + repeat * repeatDuration;
                     var distanceProgress = d / length;
 
+                    if (sliderStart)
+                        distanceProgress = 0;
```
Comment (peppy) [inline osu.Game.Rulesets.Osu/Objects/Slider.cs]: why is this needed?

Comment (Shawdooow) [reply osu.Game.Rulesets.Osu/Objects/Slider.cs]: this makes even numbered repeats where the start circle is like normal sliders, before they were always at the end circle

**Comment by smoogipoo** [comment]
@Shawdooow oops I kind of abused this branch, sry. @peppy I don't think I'm eligible to approve this anymore :P

**Review by peppy — APPROVED** [review]
